specification/sections/01-flag-evaluation.md

```diff
@@ -31,6 +33,20 @@ OpenFeature.setProvider(new MyProvider());
 
 See [provider](./02-providers.md) for details.
 
+#### Requirement 1.1.2.2
+
+> The `provider mutator` function **MUST** invoke the `initialize` function on the newly registered provider before using it to resolve flag values.
```

Comment:
Do you think we should specify an initialization timeout? I'm sure this could be handled in a provider or userland but it may be worth explicitly defining the maximum amount of time the initialization process will wait.

Reply:
Do we want to clarify what the expected order is? Do we want to call `shutdown` on the previous provider before calling `initialise` on the new provider or the other way around?

Reply:
@weyert I've tried to cover that [here](https://github.com/open-feature/spec/pull/179/files#diff-1443054d965453435ba3da2907dabbc282d2f808c3a64800f10e05fab8351b1aR11-R27).

Basically, this means that the system should never use a provider that hasn't been initialized or shutdown.

Reply:
What should the behavior be if someone tries to resolve flag values before the provider is in a ready state?

Reply:
I think the default value should be returned with the [error code](https://docs.openfeature.dev/specification/types#error-code) `PROVIDER_NOT_READY`.

Reply:
> I think the default value should be returned with the [error code](https://docs.openfeature.dev/specification/types#error-code) `PROVIDER_NOT_READY`.

This is my feeling as well.

Reply:
Added, see [this commit](https://github.com/open-feature/spec/pull/179/commits/164071b5a1ad19553c19156c8a0b68521a9324d9) for the delta. I hope it's clear enough.

@Kavindu-Dodan @justinabrahms